apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.usecase.ts

```diff
@@ -118,25 +118,55 @@ export class UpdateSubscriberChannel {
       return;
     }
 
+    let deviceTokens: string[] = [];
+
+    if (updatePayload.credentials?.deviceTokens) {
+      deviceTokens = this.unionDeviceTokens(
+        existingChannel.credentials.deviceTokens ?? [],
+        updatePayload.credentials.deviceTokens
+      );
+    }
+
     await this.invalidateCache.invalidateByKey({
       key: buildSubscriberKey({
         subscriberId: foundSubscriber.subscriberId,
         _environmentId: foundSubscriber._environmentId,
       }),
     });
 
-    const mergedChannel = Object.assign(existingChannel, updatePayload);
+    const mappedChannel: IChannelSettings = this.mapChannel(updatePayload, existingChannel, deviceTokens);
 
     await this.subscriberRepository.update(
       {
         _environmentId: environmentId,
         _id: foundSubscriber,
         'channels._integrationId': existingChannel._integrationId,
       },
-      { $set: { 'channels.$': mergedChannel } }
+      { $set: { 'channels.$': mappedChannel } }
     );
   }
 
+  private mapChannel(
+    updatePayload: Partial<IChannelSettings>,
+    existingChannel: IChannelSettings,
+    deviceTokens: string[]
+  ): IChannelSettings {
+    return {
+      _integrationId: updatePayload._integrationId || existingChannel._integrationId,
+      providerId: updatePayload.providerId || existingChannel.providerId,
+      credentials: { ...existingChannel.credentials, ...updatePayload.credentials, ...{ deviceTokens } },
```

Comment:
```suggestion
      credentials: { ...existingChannel.credentials, ...updatePayload.credentials, deviceTokens },
```
I believe that should have the same result

Reply:
TRUE :)